src/main/java/com/example/wdmsystem/reservation/system/ReservationService.java

```diff
@@ -46,4 +92,12 @@ public void cancelReservation(int reservationId) {
         reservationRepository.save(reservation);
     }
 
+    public void deleteReservation(int reservationId) {
```

Comment:
I think having a delete won't be bad. I could restrict access so that only admin can delete, so that we would still keep the logic described in the doc.